hs_discover/src/components/Resources.vue

```diff
@@ -0,0 +1,752 @@
+      },
+    };
+  },
+  name: 'Resources',
+  props:
+  ['columns', 'labels'],
+  components: {
+    datePick: DatePick,
+  },
+  computed: {
+    filteredResources() {
+      // this routine typically completes in thousandths or hundredths of seconds with 3000 resources in Chrome
+      let resfiltered = this.resources;
+      if (this.uidFilter.length === 0 && this.authorFilter.length === 0 && this.ownerFilter.length === 0
+          && this.subjectFilter.length === 0 && this.availabilityFilter.length === 0 && this.contributorFilter.length
+          === 0 && this.typeFilter.length === 0) {
+        // do nothing
+      } else {
+        // Filters should be most restrictive when two conflicting states are selected
+        if (this.uidFilter.length > 0) {
+          const resUids = resfiltered.filter(element => this.uidFilter.indexOf(element.short_id) > -1);
+          resfiltered = resUids;
+        }
+        if (this.authorFilter.length > 0) {
+          const resAuthors = resfiltered.filter(element => this.authorFilter.indexOf(element.author) > -1);
+          resfiltered = resAuthors;
+        }
+        if (this.ownerFilter.length > 0) {
+          const resOwners = resfiltered.filter(res => res.owner.filter(val => this.ownerFilter.includes(val))
+            .length > 0);
+          resfiltered = resOwners;
+        }
+        if (this.subjectFilter.length > 0) {
+          const resSubjects = resfiltered.filter(res => res.subject.filter(val => this.subjectFilter.includes(val))
+            .length > 0);
+          resfiltered = resSubjects;
+        }
+        if (this.availabilityFilter.length > 0) {
+          const resAvailabilities = resfiltered.filter(res => res.availability
+            .filter(val => this.availabilityFilter.includes(val)).length > 0);
+          resfiltered = resAvailabilities;
+        }
+        if (this.contributorFilter.length > 0) {
+          const resContributors = resfiltered.filter(res => res.contributor.filter(val => this.contributorFilter.includes(val))
+            .length > 0);
+          resfiltered = resContributors;
+        }
+
+        if (this.typeFilter.length > 0) {
+          const resTypes = resfiltered.filter(element => this.typeFilter.indexOf(element.type) > -1);
+          resfiltered = resTypes;
+        }
+      }
+      if (this.startdate !== 'Start Date' && this.startdate !== '' && this.enddate !== 'End Date' && this.enddate !== '') {
+        const resDate = [];
+        resfiltered.forEach((item) => {
+          if (item.start_date && item.end_date) {
+            if (this.dateOverlap(item.start_date, item.end_date)) {
+              resDate.push(item);
+            }
+          }
+        });
+        resfiltered = resDate;
+      }
+      resfiltered = this.columnSort(resfiltered);
+      return resfiltered;
+    },
+  },
+  watch: {
+    resources() {
+      this.populateFilters();
+      if (this.resources.length > 0) {
+        this.resloaded = true;
```

Comment:
Not sure what is trying to be done with resloaded.  Aren't all resources served with the page?  So the only way resloaded would be False is if there were no discoverable resources.

Reply:
hmm, there seems to be more to this.  Still looking over everything.

Reply:
ok, so there are two different calls.  SearchView and SearchAPI.  SearchView serves the page while SearchAPI is called by the vue page to populate resources.

Reply:
And the only filter passed to the api is a content search.  Looks like all filtering after that happens client side.

Reply:
Playing around on dev-hs-5.  When navigating to the discover page, there are 2 hits to discoverapi.  The first hit has a cache parameter which just limits the results to 120 resources.  It takes 241 ms to get a response and 80 ms to receive.  These results are parsed and loaded shortly after.

The second request makes a full query of all resources.  This takes 2 seconds to respond and another 2-3 seconds to transfer the nearly 6 MB response.  The vue discover page takes about 5 seconds to parse and display the results of the response.  In all it takes 10 seconds to load discover fully.

Reply:
While vue waits for the second response, the mouse is overtaken by a loading icon.  I can still select filters though.  However, when vue completes loading the second response (after 10 seconds) any filters I make are cleared.  I don't think the cached response is really buying us much.

Reply:
dev-hs-5 probably not the best place to do functional testing. I should have clarified this is code review and familiarization phase and that I will later post a link to an environment for functional/UI testing. Hope you're able to attend the feedback session as part of tomorrow's Wed call.

Reply:
I'll be at the Wednesday call. The 5 seconds client side will be the same regardless of the environment.  Considering you added the cache call, it seems you have some concerns about response time already.

Reply:
I think you're calculating things a different way. The load you are observing is for the async map call, not until all resources are actionable. I'll explain later, but the client load is hundredths of a second for filter construction, then some time rendering the filter checkboxes which is under negotiation to truncate. Still, you're counting the map load, not the render.

Reply:
The reason this takes so long is because of the requirement to have checkbox filters for so many items. Since we aim to split Enhancement and design conversations from the bugfix release, this is temporary until those get removed or at least redesigned and reimplemented. Once that happens, the load times are normal; we're suffering load time penalties, as we are required to carry forward the legacy design and functionality of the checkboxes. Also, there are SOLR optimizations underway that will likely coincide with the Discover bugfix release and shorten the 5 second server-side. Did you see the devtools console times for the frontend calls or were you looking only at the network?